include/swift/Syntax/SyntaxArena.h

```diff
@@ -31,6 +32,14 @@ class SyntaxArena : public llvm::ThreadSafeRefCountedBase<SyntaxArena> {
 
   llvm::BumpPtrAllocator Allocator;
 
+  /// \c SyntaxArenas that are kept alive as long as this arena is alive.
+  /// If a \c RawSyntax A node in this arena references a \c RawSyntax node B
+  /// from a different arena, the B's arena will be added to this list.
+  /// The \c SyntaxArena's in this set are manually retained when added to this
+  /// list by \c addChildArena and manually released in the destructor of this
+  /// arena.
+  llvm::SmallPtrSet<SyntaxArena *, 4> ChildArenas;
```

Comment:
No, there’s not. But I can’t think of a real-world scenario where a circular reference might occur since newer arenas only ever reference older arenas (older = from a previous parsing run). Might be worth adding a check in assert builds though, just to be safe for the future.